File: Java/Sub-Arrays/Kadanes_Algorithm.java
import java.util.Scanner;

public class Kadanes_Algorithm {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the size of Array : ");
        int n = sc.nextInt();
        int [] arr = new int [n];
        System.out.println("Enter the elements of the array : ");
        for (int i = 0; i < n; i++) {
            arr[i]= sc.nextInt();
        }
        System.out.println("Maximum Sub-array sum is "+kadane(arr , n));
    }
    static int kadane(int [] nums , int n){
        int max = Integer.MIN_VALUE, sum = 0;
        for(int i=0;i<n;i++){
            sum += nums[i];
            max = Math.max(sum,max);

            if(sum<0) sum = 0;
        }
        return max;
    }
}

File: C/Sub-Arrays/Kadanes Algorithm.c
#include <stdio.h>

// macro defining maximum capacity of the array
#define MAX 100

// this function is returning the value which is greater in between a and b.
int max_getter(int a, int b)
{
    return a >= b ? a : b;
}

// returning the maximum subarray sum
int kadanes_algo(int n, int arr[])
{
    // maximum subarray sum is denoted by max_sum
    int max_sum = -1e9;

    // sum so far is stored in curr_sum
    int curr_sum = 0;

    for (int i = 0; i < n; ++i)
    {
        // adding arr[i] to the curr_sum
        curr_sum += arr[i];

        // if max_sum is greater than curr_sum then we are updating max_sum
        max_sum = max_getter(curr_sum, max_sum);

        // if curr_sum < 0 we are making it 0 again.
        if (curr_sum < 0)
        {
            curr_sum = 0;
        }
    }

    // retunrning maximum subarray sum
    return max_sum;
}

int main()
{
    // declaring n -> number of elements in the array and array having capacity of MAX elements
    int n, arr[MAX];
    printf("Enter the number of elements in the array\n");
    // taking input the number of elements
    scanf("%d", &n);

    // taking input the elemnets of the array
    printf("Enter the elements in the array\n");
    for (int i = 0; i < n; ++i)
    {
        scanf("%d", &arr[i]);
    }

    // calling function kadanes_algo to get maximum subarray sum
    printf("Maximum subarray sum is: %d", kadanes_algo(n, arr));

    return 0;
}
